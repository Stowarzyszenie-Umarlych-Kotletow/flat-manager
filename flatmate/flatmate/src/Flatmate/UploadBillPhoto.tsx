import {Modal, ModalContent, ModalTitle} from "react-native-modals";
import {Image, TextInput, View} from "react-native";
import styles from "../static/styles";
import {Button} from "react-native-elements";
import * as React from "react";
import { useState, useEffect } from "react";
import * as ImagePicker from 'expo-image-picker';
import {Controller, useForm} from "react-hook-form";

export function UploadBillModal ({ setShowUploadBillModal}) {
    const [imageUri, setImageUri] = useState(null);
    async function addImage() {
        let image_data = await ImagePicker.launchImageLibraryAsync({
            mediaTypes: ImagePicker.MediaTypeOptions.Images,
            allowsEditing: true,
            }
        );
        if (!image_data["cancelled"]) {
            setImageUri(image_data["uri"])
        }
    };

    function handleUploadPhoto(data: CreateFlatRequest) {
        // backend connection
        console.log(data.name);
        console.log(imageUri);
    }

    const {control, handleSubmit, formState: {errors}} = useForm({
        defaultValues: {
            name: '',
        },
    });

    function uploadPhoto() {
    }

    const  checkForCameraRollPermission=async()=>{
        const { status } = await ImagePicker.getMediaLibraryPermissionsAsync();
        if (status !== 'granted') {
          alert("Please grant camera roll permissions inside your system's settings");
        }else{
          console.log('Media Permissions are granted')
        }
    } 

    useEffect(() => {
        checkForCameraRollPermission()
      }, []);

    return (
        <Modal
            width={0.9}
            rounded
            actionsBordered
            style={{zIndex: 1000}}
            visible={true}
            modalTitle={<ModalTitle title="Upload Bill"  align="left" />}
            onTouchOutside={() => { setShowUploadBillModal(false)}}
        >
            <ModalContent>
                <View style={styles.imageUploadContainer}>
                    {
                    imageUri  && <Image source={{ uri: imageUri }} style={{ width: 400, height: 400 }} />
                    }
                    <Button
                        buttonStyle={styles.blueButton}
                        title={imageUri ? 'Change Image' : 'Choose Image'}
                        onPress={addImage}
                    />
                </View>
                <Controller
                    control={control}
                    name="name"
                    rules={{
                    maxLength: 100,
                    }}
                    render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Name"
                    />
                    )}
                />
                <Button
                    buttonStyle={styles.greenButton}
                    title="Send"
                    onPress={handleSubmit(handleUploadPhoto)}
                />
                <Button
                    buttonStyle={styles.warnButton}
                    title="Cancel"
                    onPress={() => { setShowUploadBillModal(false) }}
                />
            </ModalContent>
        </Modal>

    )
}
