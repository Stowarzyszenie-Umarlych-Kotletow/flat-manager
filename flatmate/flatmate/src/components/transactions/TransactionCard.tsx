import * as React from "react";
import {useState} from "react";
import styles from "../../static/styles";
import {Text, View, TouchableOpacity} from "react-native";
import {TransactionDetailsModal} from "./TransactionDetailsModal";
import {useFlat} from "../../features/hooks";
import {formatDate} from "../../helpers/date-helper";
import {useDeleteTransactionGroupMutation} from "../../features/api/transaction-api";

const deleteIcon = require("../../static/taskDelete.svg") as string;

export function TransactionCard({transactionGroup}) {
  const {flat, flatId, flatTasks, flatUsers} = useFlat();
  const [showTransactionDetails, setShowTransactionDetails] = useState(false);
  const [deleteTransactionGroup] = useDeleteTransactionGroupMutation();

  function handleDeleteTransactionGroup(){
    const dataDict = {
      transactionGroupId: transactionGroup.id,
    };
    deleteTransactionGroup(dataDict)
  }

  function getUsername(userId: string): string {
    for (const user of flatUsers) {
        if(userId == user.id)
            return user.username;
    }
    return "Unknown user";
  }

  return (
    <TouchableOpacity
      style={styles.card}
      onPress={() => {
        setShowTransactionDetails(true)
      }}
    >
      <View style={styles.viewRow}>
        <Text style={styles.cardTitle}>{transactionGroup.name}</Text>
        <Text style={styles.cardText}>{formatDate(transactionGroup.dateCreated)}</Text>
      </View>

      <View style={styles.viewRow}>
        <Text style={styles.cardText}>{getUsername(transactionGroup["createdBy"])}</Text>
        <Text style={styles.cardText}>{transactionGroup["total"]}PLN</Text>
      </View>

      <View style={styles.viewRow}>
        <TouchableOpacity
          onPress={() => {
            handleDeleteTransactionGroup();
          }}
          style={{marginStart: 10}}
        >
          <img src={deleteIcon} alt="taskDeleteIcon" style={{width: '20px', height: '20px'}}/>
        </TouchableOpacity>
      </View>
      {showTransactionDetails ? (
        <TransactionDetailsModal setShowTransactionDetailsModal={setShowTransactionDetails}
                                 transactionGroup={transactionGroup}/>
      ) : null}
    </TouchableOpacity>
  );
};