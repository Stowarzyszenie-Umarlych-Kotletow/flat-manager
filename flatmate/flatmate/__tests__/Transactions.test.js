import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mockStore } from "./utils/test-utils";
import store from "../src/store";
import { ModalPortal } from 'react-native-modals';
import { CreateTransactionGroup } from '../src/Transactions/CreateTransactionGroup';
import { TransactionCard } from '../src/Transactions/TransactionCard';
import { TransactionManagementView } from '../src/Transactions/TransactionManagementView';
import { UploadBillModal } from '../src/Transactions/UploadBillPhoto';


let transactionGroup = { 
  id: 1,
  paid_by: 123,
  total: 45,
  date: '21-01-2022',
  title: 'Zakupy Carefour',
  transactions: [
    {
      id: 1,
      name: "ciasteczka",
      total: 10,
      shares: [
        {id: 123, percentage : 60, resolved: true},
        {id: 124, percentage : 40, resolved: false},
      ],
    }
  ],
  participants: [
    123, 122
  ]
};

test('CreateTransactionGroup Screen renders correctly', () => {
  const tree = renderer.create(<Provider store={mockStore()}><CreateTransactionGroup /><ModalPortal/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});


test('TransactionCard Screen renders correctly', () => {
  const tree = renderer.create(<Provider store={mockStore()}><TransactionCard transactionGroup={{transactionGroup}} /><ModalPortal/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('TransactionManagementView Screen renders correctly', () => {
  const tree = renderer.create(<Provider store={mockStore()}><TransactionManagementView /><ModalPortal/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('UploadBillModal Screen renders correctly', () => {
  const tree = renderer.create(<Provider store={mockStore()}><UploadBillModal /><ModalPortal/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});