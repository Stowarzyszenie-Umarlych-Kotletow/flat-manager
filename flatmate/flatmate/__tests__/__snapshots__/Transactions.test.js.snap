// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTransactionGroup Screen renders correctly 1`] = `
<View
  pointerEvents="auto"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      false,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 10,
        "paddingRight": 10,
        "zIndex": 1000,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "#000",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "left": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 8,
            "height": null,
            "opacity": 0,
            "overflow": "hidden",
            "width": 675,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "padding": 14,
                "paddingHorizontal": 18,
              },
              Object {
                "alignItems": "flex-start",
              },
              Object {
                "backgroundColor": "#F9F9FB",
                "borderBottomWidth": 0.5,
                "borderColor": "#DAD9DC",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151822",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
                undefined,
              ]
            }
          >
            Add Transaction Group
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 24,
              },
              Object {
                "paddingTop": 0,
              },
              undefined,
            ]
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
               Set Transaction Group Title 
            </Text>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Transaction Group Name"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#fafafa",
                  "borderColor": "#eaeaea",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "fontSize": 14,
                  "height": 43,
                  "marginBottom": 5,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                  "paddingLeft": 10,
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
             Items 
          </Text>
          <Text>
             No Items Yet 
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
             Assign Users to Transaction Group 
          </Text>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "height": 55,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "marginLeft": 15,
                    "marginTop": 15,
                  }
                }
              >
                <Text />
              </View>
              <TextInput
                allowFontScaling={true}
                clearButtonMode="always"
                onChangeText={[Function]}
                placeholder="Search"
                placeholderTextColor="#757575"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#00a2dd",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "height": 35,
                    "margin": 0,
                    "marginLeft": -25,
                    "marginTop": 10,
                    "padding": 5,
                    "paddingLeft": 30,
                    "width": 730,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <RCTScrollView
              style={
                Object {
                  "height": 130,
                  "padding": 5,
                }
              }
            >
              <View />
            </RCTScrollView>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#3897f1",
                      "borderColor": "#2089dc",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginRight": 25,
                      "marginTop": 10,
                      "padding": 8,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Add Item
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#3897f1",
                      "borderColor": "#2089dc",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginRight": 25,
                      "marginTop": 10,
                      "padding": 8,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Upload Bill
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#5cd07f",
                      "borderColor": "#2089dc",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginRight": 25,
                      "marginTop": 10,
                      "padding": 8,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Submit
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#e74040",
                      "borderColor": "#2089dc",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginRight": 25,
                      "marginTop": 10,
                      "padding": 8,
                      "width": 200,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`TransactionCard Screen renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "rgb(0, 102, 255)",
      "borderRadius": 5,
      "borderWidth": 1,
      "marginEnd": 20,
      "marginStart": 20,
      "marginTop": 20,
      "opacity": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Name not found
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      PLN
    </Text>
  </View>
</View>
`;

exports[`TransactionManagementView Screen renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "calc(100vh - 75px)",
      "overflow": "scroll",
      "paddingBottom": "20px",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "800",
        "marginBottom": 30,
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    Manage transactions
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5cd07f",
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Create Transaction Group
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "rgb(0, 102, 255)",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginEnd": 20,
        "marginStart": 20,
        "marginTop": 20,
        "opacity": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Zakupy Carefour
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        21-01-2022
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Adam
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        45
        PLN
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "rgb(0, 102, 255)",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginEnd": 20,
        "marginStart": 20,
        "marginTop": 20,
        "opacity": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Wyjście do restauracji
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        20-01-2022
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        Edward
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "marginBottom": 5,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
        427
        PLN
      </Text>
    </View>
  </View>
</View>
`;

exports[`UploadBillModal Screen renders correctly 1`] = `
<View
  pointerEvents="auto"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      false,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 1000,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "#000",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "left": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 8,
            "height": null,
            "opacity": 0,
            "overflow": "hidden",
            "width": 675,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "padding": 14,
                "paddingHorizontal": 18,
              },
              Object {
                "alignItems": "flex-start",
              },
              Object {
                "backgroundColor": "#F9F9FB",
                "borderBottomWidth": 0.5,
                "borderColor": "#DAD9DC",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#151822",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
                undefined,
              ]
            }
          >
            Upload Bill
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 18,
                "paddingVertical": 24,
              },
              Object {
                "paddingTop": 0,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "justifyContent": "center",
                "padding": 50,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#3897f1",
                      "borderColor": "#2089dc",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "center",
                      "marginLeft": 25,
                      "marginRight": 25,
                      "marginTop": 10,
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Choose Image
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#5cd07f",
                    "borderColor": "#2089dc",
                    "borderRadius": 5,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginRight": 25,
                    "marginTop": 10,
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Send
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#e74040",
                    "borderColor": "#2089dc",
                    "borderRadius": 5,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "marginLeft": 25,
                    "marginRight": 25,
                    "marginTop": 10,
                    "padding": 8,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
